.github/workflows/build.yml

```diff
@@ -0,0 +1,80 @@
+name: build
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  build:
+    name: ${{ matrix.os }} / GHC ${{ matrix.ghc }}
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      matrix:
+        os:
+          - ubuntu-16.04 # https://launchpad.net/~hvr/+archive/ubuntu/ghc?field.series_filter=xenial
+        ghc:
+          - 7.0.4
+          - 7.2.2
+          - 7.4.2
+          - 7.6.3
+          - 7.8.4
+          - '7.10'
+          - '8.0'
+          - '8.2'
+          - '8.4'
+          - '8.6'
+          - '8.8'
+          - '8.10'
+    steps:
+      - uses: actions/checkout@v2
+      - uses: haskell/actions/setup@v1
+        with:
+          ghc-version: ${{ matrix.ghc }}
+
+      - name: install typediff
+        run: |
+          mkdir -p "$HOME/.local/bin"
+          curl -L https://github.com/haskell-compat/base-compat/releases/download/typediff-0.1.4/typediff > "$HOME/.local/bin/typediff"
+          chmod a+x "$HOME/.local/bin/typediff"
+          echo "$HOME/.local/bin" >> "$GITHUB_PATH"
+
+      - run: cabal update
+
+      - name: enable documentation
+        run: echo "enable_documentation=--enable-documentation" >> $GITHUB_ENV
+        if: matrix.ghc != '7.2.2'
```

Comment:
@RyanGlScott building docs fails on 7.2.2.

Reply:
Good catch. This is one of the hacks I put together in https://github.com/haskell-compat/base-compat/issues/65#issuecomment-566665746 that I had forgotten about.

On a related note, I'm somewhat worried that `cabal build --enable-documentation` is silently ignoring errors that may arise when building documentation. [Here](https://github.com/haskell-compat/base-compat/pull/72/checks?check_run_id=1820388783#step:8:542) is a concerning bit from the GHC 7.6 log:

```
Warning: Failed to build documentation for mtl-2.2.2 (which is required by
test:type-check-test from type-check-0.1.0, exe:dumptypes from
type-check-0.1.0 and others).
```

As a result, it appears that this never builds the documentation for `base-compat-batteries` at all. Perhaps we should be running `cabal haddock` instead?